distributed_training/validator/reward.py

```diff
@@ -260,13 +267,12 @@ async def get_rewards(
         # Set up the scores tensor
         scores = torch.FloatTensor([1 for _ in self.miner_uids]).to(self.device)
 
-        # Update mapping of uids to peerids
-        self.uids_to_peerids = await map_uid_to_peerid(self, range(0, self.metagraph.n))
-        self.uids_to_peerids[self.uid] = self.dht.peer_id
-        bt.logging.info(f"UID To PeerID Mapping: {self.uids_to_peerids}")
-
         # Check if peer is connected to DHT & run_id and blacklist them if they are not
-        blacklist_scores = await score_blacklist(self, self.miner_uids.tolist())
+        bt.logging.info(f"UID To PeerID Mapping: {self.uids_to_peerids}")
+        update_run_peerid_list(self)
+        # Update peerids from chain
+        map_uid_to_peerid(self)
```

Comment:
Given this is related to my original PR and in the spirit of getting this tested sooner rather than later I can pick this up whilst you finish any of the other tasks and BT testing.

My original thinking for doing this as a background task was to avoid it slowing down the main validator process. If the background thread causes errors then we can just loop around the `self.miner_uids` for that loop.